Source/Fields.H

```diff
@@ -0,0 +1,131 @@
+/* Copyright 2024 Luca Fedeli
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef WARPX_FIELDS_H_
+#define WARPX_FIELDS_H_
+
+#include <ablastr/fields/MultiFabRegister.H>
+
+#include <AMReX_Enum.H>
+
+#include <algorithm>
+#include <iterator>
+
+
+namespace warpx::fields
+{
+    AMREX_ENUM(FieldType,
+        None,
+        Efield_aux,
+        Bfield_aux,
+        Efield_fp,
+        Bfield_fp,
+        Efield_fp_external,
+        Bfield_fp_external,
```

Comment:
Down the road, removing the `field` from the `Bfield` and `Efield` names would be sensible for consistency.